content/blog/2020-08-14-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,242 @@
+
+First we'll create a separate sheet for our pivot table, specifying the range of the data we want it to be based on. 
+
+![Pivot set up](screenshots/pivot-set-up.png)
+
+And then, it's as simple as dragging and dropping our variables of interest into specific order with `expense` and `quarter` in rows and `budget` and `amount` in values, specifying a *sum* calculation.
+
+![Pivot table](screenshots/pivot-table.png)
+
+
+## Row calculations
+
+One thing that can be tricky when working with pivot tables is calculating new variables by rows as opposed to columns. Coming back to our finance example, let's say we want to examine how much money is left over from each quarter based off of budget and amount spent.  A quick work around could be to manually add in a formula in a cell adjacent to our table and drag it down to each respective row, a process depicted below.
+
+![Left over](screenshots/left-over-calculation.png)
+
+Working with our data in R we can take advantage of the `rowwise()` function from dplyr, telling R we want to compute new values and summary statistics by row and not column.  
+Let's take a look at how much money was left over in each quarter by subtracting money spent from our budget and use the `rowwise()` function to specify we want this done by each row in our summary table.
+
+```{r}
+left_over <- spending_table %>%
+  rowwise() %>%
+  mutate(left_over = total_budget_amt - total_expense_amt) %>%
+  arrange(left_over)
+
+left_over
+```
+
+Great! Using the `arrange()` function we can also bring the largest instances of overspending to the top of our table to focus our attention.
+
+We can also filter our table if we're interested in a specific expense type. Let's hone in on spending for office happy hours.
+
+```{r}
+left_over %>%
+  filter(expense == "office happy hour")
+```
+
+Hmm, a lot of *over*spending in Quarter 3, which might make sense as these are the summer months. 
+
+## From tables to visualizations
+
+Finally, we can apply data visualization to our workflow and look at spending across quarters and expense type. We will use the ggplot2 package for visualisation, which already got loaded when we called `library(tidyverse)` and we'll also load the scales package so that we can display the values on the y-axis in dollar format.
+
+```{r,message=FALSE}
+library(scales)
+
+left_over %>%
+  ggplot(aes(quarter, left_over)) +
+  geom_point() +
+  geom_line(group = 1) +
+  facet_wrap(~expense) +
+  scale_y_continuous(labels = dollar) +
+  geom_hline(yintercept = 20, color = "red") +
+  labs(
+    title = "Spending by expense type",
+    x = "Quarter",
+    y = "Amount jeft over, in $"
+  )
+```
+
+Voila! We have both a summary table and a graphic that can tell us useful information about company spending. Hmm, there might be something to the trend we see with the overspending on happy hours in Q3 after Q1 and Q2 were sunk with lawsuits.
```

Comment:
Do we have any observations about the R plotting process versus what one would do in Excel? I suspect faceting is a good example of something that's not possible (or this easy) in Excel.

Reply:
Perhaps we should remove the plot all together? By removing the rowwise section this part doesn't make sense (we even lose the left_over dataframe). I also rarely make graphs in Excel so I don't think a direct comparison here would be as useful.

Reply:
I think the post is pretty long, so eliminating the plot and this section sounds like a good move to me.

Reply:
Agreed - section removed